chapters/ch09.md

```diff
@@ -19,24 +27,37 @@ Suppose you care about both the accuracy and the running time of a learning algo
 It seems unnatural to derive a single metric by putting accuracy and running time into a single formula, such as:
 
 ->
+| Classifier  | Độ chính xác | Thời gian chạy |
+| ----- | -------: | -------: |
+| A  | 90%  | 80ms |
+| B  | 92%  | 95ms |
+| C  | 95%  | 1,500ms |
+
+Việc tạo ra một phép đo đơn nhất bằng cách đưa cả độ chính xác và thời gian chạy vào một công thức có vẻ không tự nhiên, ví dụ như:
+
 
 Accuracy - 0.5*RunningTime
 
 ->
+Độ chính xác - 0.5*(Thời gian chạy)
 
 Here’s what you can do instead: First, define what is an “acceptable” running time. Lets say anything that runs in 100ms is acceptable. Then, maximize accuracy, subject to your classifier meeting the running time criteria. Here, running time is a “satisficing metric”—your classifier just has to be “good enough” on this metric, in the sense that it should take at most 100ms. Accuracy is the “optimizing metric.”
 
 ->
+Thay vào đó, bạn có thể làm như sau: Trước hết, định nghĩa thế nào là một mốc thời gian chạy "chấp nhận được". Giả sử mốc dưới 100ms là chấp nhận được. Sau đó, hãy cực đại hóa độ chính xác, với ràng buộc là classifier đó vẫn đảm bảo yêu cầu về thời gian chạy. Ở đây, thời gian chạy là một "phép đo đủ đạt yêu cầu" - classifier của bạn chỉ cần "đủ tốt" về mặt này (i.e. thời gian), theo nghĩa nó chỉ được phép chạy dưới 100ms. Độ chính xác mới là "phép đo tối ưu".
 
 If you are trading off N different criteria, such as binary file size of the model (which is important for mobile apps, since users don’t want to download large apps), running time, and accuracy, you might consider setting N-1 of the criteria as “satisficing” metrics. I.e., you simply require that they meet a certain value. Then define the final one as the “optimizing” metric. For example, set a threshold for what is acceptable for binary file size and running time, and try to optimize accuracy given those constraints.
 
 ->
+Nếu như bạn phải cân bằng giữa N tiêu chí khác nhau, VD như kích thước của file nhị phân của model (quan trọng với các mobile app, vì người dùng không muốn download những app nặng), thời gian chạy, và độ chính xác, bạn có thể cân nhắc đặt N-1 trong số các tiêu chí là các tiêu chí "vừa đủ". Có nghĩa bạn chỉ cần yêu cầu chúng đạt một số giá trị nào đó. Sau đó hãy định nghĩa tiêu chí cuối cùng như là tiêu chí "tối ưu". Ví dụ, đặt một mức ngưỡng thế nào là chấp nhận được với một file nhị phân và thời gian chạy, và sau đó hãy cố gắng tối ưu hóa độ chính xác với những điều kiện ràng buọc đã đưa ra.
```

Comment:
> Nếu như ~~bạn~~ phải cân bằng giữa N tiêu chí khác nhau, ~~VD~~ ví dụ như kích thước file lưu trữ của ~~của nhị phân của model~~ ( _điều rất_ quan trọng với các ứng dụng mobile ~~mobile app~~, vì người dùng không muốn ~~download~~ tải về những ~~app~~ ứng dụng nặng), thời gian chạy, độ chính xác, _thì_ bạn có thể cân nhắc đặt N-1 trong số các tiêu chí là các tiêu chí "vừa đủ". 

- Không nên viết tắt (VD), và dịch toàn diện các từ tiếng anh khác (download)
- kích thước nhị phân kia là ám chỉ dung lượng lưu trữ của ứng dụng. 
- từ model cần được thảo luận ở glossary 

> Có nghĩa bạn chỉ cần yêu cầu **chúng** đạt một số giá trị nào đó. Sau đó hãy định nghĩa tiêu chí cuối cùng như là tiêu chí "tối ưu".
- chúng ở đây là mô hình học máy --> cần chọn từ rõ nghĩa hơn

> Ví dụ, đặt một mức ngưỡng thế nào là chấp nhận được với một file nhị phân và thời gian chạy, và sau đó hãy cố gắng tối ưu hóa độ chính xác với những điều kiện ràng ~~buọc~~ buộc ~~đã~~ đưa ra.
- tương tự với định nghĩa file nhị phân ở trên --> dung lượng mô hình

Reply:
Ở đây là quan điểm dịch "sát" đến đâu so với bản gốc. Từ binary file lúc đầu mình đã dịch là file chạy hoặc là file thực thi vì thường khi nghĩ đến binary file ta nghĩ đến file thực thi, VD như file .exe trong Windows OS. Nhưng ngay sau đó mình thấy view đó là quá hẹp, file chạy là executable file còn file nhị phân nghĩa rộng hơn nhiều (file apk, file ảnh, vv...) Vậy tốt nhất là để như nguyên gốc: binary file = file nhị phân. File nhị phân được định nghĩa là non-text file.

Vậy, nếu lấy tinh thần bám sát nguyên tác, thì giữ nguyên. Còn nếu muốn dịch thêm cho khỏi cứng, thì có thể thay bằng "(...) N tiêu chí khác nhau, ví dụ như kích thước thành phẩm của ứng dụng". 

P.S mô hình đâu có được stored ở dạng binary file?